modules/gatsby/src/gatsby-node/source-nodes.js

```diff
@@ -117,6 +117,10 @@ function sourceNodes({actions}) {
       chapters: [lvl1Chapter]
       entries: [Entry]
     }
+
+    type ImageSharp implements Node {
+      id: ID!
```

Comment:
What does the exclamation markd do? Only used for this type... 

This is a `.js` file, so I assume you could add some comments above the string and let others know.

Reply:
The exclamation mark means the id can be empty.